src/components/menu/position/MenuPosition.tsx

```diff
@@ -0,0 +1,78 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import MenuContentPositioned from './MenuContentPositioned';
+import { IMenuPositionConfig } from '../MenuSharedTypes';
+import RenderChildrenInPortal from '../utils/RenderChildrenInPortal';
+
+export interface IMenuPositionProps extends IMenuPositionConfig {
+    opened: boolean;
+    topLevel: boolean;
+    portalTarget: Element;
+    contentWrapper?: (props: { children: React.ReactNode }) => JSX.Element;
+    toggler: React.ReactNode;
+    children: React.ReactNode;
+}
+
+export interface IMenuPositionState {
+    elTogglerInitialized: boolean;
+}
+
+export default class MenuPosition extends React.Component<IMenuPositionProps, IMenuPositionState> {
+    public static defaultProps = {
+        contentWrapper: React.Fragment
+    };
+
+    public state = {
+        elTogglerInitialized: false
+    };
+
+    private elToggler: HTMLElement = null;
+
+    // React Measure is not used because it cannot detect the left/top coordinate
+    // changes of absolute positioned blocks. This caused problems where left/top
+    // positions from React Measure were outdated. To solve this we do the
+    // measurements manually in MenuContentPositioned at the correct time.
+
+    public render() {
+        const MaybePortal = this.props.topLevel
```

Comment:
Could you please explain or rename this variable `MaybePortal`? Its not quite clear why its there

Reply:
Renamed `PortalIfTopLevelMenu` and added comment